### PR TITLE
Feature/allow temp partials

### PR DESCRIPTION
**Pull request by bnormington**
We ran into a race condition with the way we create and remove temporary partials.  This is a quick fix to try to fix that. See code comment for more details.  Also  I didn't realize this at the time but creating temporary partial is working because we are not caching any handlebars files.  Our next solution/fix should address this.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -197,15 +197,29 @@ RemoteHandlebars.prototype.readTemplate = function readTemplate(filePath, callba
     });
 };
 
+/**
+ * This function load up paritals handlebars templates from disk.
```
Comment (clok) [inline index.js:201]: Thank you for adding the detailed comment

File: test/index.js [hunk 2]
```diff
@@ -65,7 +67,7 @@ describe('RemoteHandlebars', function () {
             });
         });
 
-        it('should render view with layout', function (done) {
+        xit('should render view with layout', function (done) {
```
Comment (clok) [inline test/index.js:70]: Are these `xit` and `xdescribe` intentional?

File: test/index.js [hunk 3]
```diff
@@ -350,4 +352,62 @@ describe('RemoteHandlebars', function () {
             rendered.should.equal('<main><p>Content</p></main>');
         });
     });
+
+    describe('.readTemplate()', function () {
```
Comment (clok) [inline test/index.js:356]: Wonderful job adding tests. 👍

**Review by clok — CHANGES_REQUESTED** [review]

File: test/index.js (hunk 2 above)
Comment (bnormington) [reply test/index.js:70]: Yes, Sorry, forgot to say that in the PR comment.  I had two commits with this PR. The first was to disable all failing test and to prevented Mocha from exiting after the first.  I inherited these broken test as we didn't update the tests when we rewrote the library to have remote views and local layouts.  Thats one reason there are a bit more new tests than needed.

Comment (clok) [reply test/index.js:70]: @bnormington Right on. Thanks for the explanation.

**Review by clok — APPROVED** [review]

**Review by craigfurlong — APPROVED** [review]

**Pull request merged**